### DISCUSSION TITLE
Getting Database Connection Strings from Azure Key Vault for Multi-Tenant Site

### DISCUSSION THREAD

**Discussion by ulieckstein** [OrchardCMS/OrchardCore#10839] [Q&A]
Hi, 
we are building a multi-tenant page, where each tenant has their own database. By default, these connection strings are stored in plain text within the appsettings.json file in the teanant folder of the storage account. This technique is a security concern to our team, so we would like to have them stored within Azure Key Vault. 

I did find a matching plugin for key vault usage, but all the examples only show usages of side-wide used keys and none for the database connection string. Is there a possibility to have the connection string stored in the key vault and have a separate connection string for every tenant? 

I do realize that this would likely break the setup process where the tenant can enter his connection string himself, but this would be okay for us as long as the connection string is safely stored.

**Comment by jtkech** [comment]
Yes, normally we can use it as a configuration source, and as from any other config source, if a tenant connection string is pre-configured it is not requested (input not shown) by the setup form. see https://github.com/OrchardCMS/OrchardCore/pull/6422#discussion_r449335833 for more details on the keys.
